### PR TITLE
Fix license format

### PR DESCRIPTION
**Pull request by PaulKMueller**
Checklist
* [x] Used a fork of the feedstock to propose changes
* [x] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/conda_smithy.html#how-to-re-render ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

The current license is not in `SPDX` format (see [here](https://spdx.org/licenses/LGPL-2.0.html) for the specification).
This PR fixes this.
Nothing huge, but should make automated license parsing tools handle this package correctly.

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -31,7 +32,7 @@ test:
 
 about:
   home: https://www.sourceware.org/pthreads-win32/
-  license: LGPL 2
+  license: LGPL-2.0
```
Comment (pavelzw) [inline recipe/meta.yaml]:
```suggestion
  license: LGPL-2.1-or-later
```

**Comment by pavelzw** [comment]
@conda-forge-admin, please rerender

File: recipe/meta.yaml
Comment (pavelzw) [inline recipe/meta.yaml:1]: we should also switch from md5 to `sha256: b9bd02958639a854461a72f72bf3e89a3c4181843b65924a54df8a50b4e5d15d`

**Comment by pavelzw** [comment]
```
Applying patch: D:\a\1\s\recipe\timespec.patch
Applying patch: D:\a\1\s\recipe\timespec.patch with args:
['-Np0', '-i', 'C:\\Users\\VSSADM~1\\AppData\\Local\\Temp\\tmp6m82_gx3\\timespec.patch.native', '--binary']
checking file 'pthreads.2\pthread.h'
Hunk #1 FAILED at 313 (different line endings).
1 out of 1 hunk FAILED
Warning: failed to download source.  If building, will try again after downloading recipe dependencies.
Error was: 
Command '['C:\\Miniforge\\Library\\usr\\bin\\patch.exe', '--no-backup-if-mismatch', '--batch', '-Np0', '-i', 'C:\\Users\\VSSADM~1\\AppData\\Local\\Temp\\tmp6m82_gx3\\timespec.patch.native', '--binary', '--dry-run']' returned non-zero exit status 1.
Traceback (most recent call last):
  File "C:\Miniforge\Scripts\conda-build-script.py", line 10, in <module>
    sys.exit(execute())
             ^^^^^^^^^
  File "C:\Miniforge\Lib\site-packages\conda_build\cli\main_build.py", line 589, in execute
    api.build(
  File "C:\Miniforge\Lib\site-packages\conda_build\api.py", line 209, in build
    return build_tree(
           ^^^^^^^^^^^
  File "C:\Miniforge\Lib\site-packages\conda_build\build.py", line 3655, in build_tree
    packages_from_this = build(
                         ^^^^^^
  File "C:\Miniforge\Lib\site-packages\conda_build\build.py", line 2446, in build
    try_download(m, no_download_source=False, raise_error=True)
  File "C:\Miniforge\Lib\site-packages\conda_build\render.py", line 779, in try_download
    raise RuntimeError(
RuntimeError: Failed to download or patch source. Please see build log for info.

```

why did the patch fail? the sources are still the same as before 🤔

**Comment by pavelzw** [comment]
@conda-forge-admin, please restart ci

**Comment by xhochy** [comment]
> why did the patch fail? the sources are still the same as before 🤔

It has been quite a while since the last build and conda-build did become a bit stricter in applying patches. You should convert to a `git format-patch` output (please ensure the authors are correctly set) and then application should work again (and keep so).

**Comment by xhochy** [comment]
This is not the output of `git format-patch --no-signature`

**Comment by pavelzw** [comment]
i think the issue was `Hunk #1 FAILED at 313 (different line endings).` which can be solved with unix2dos

**Comment by pavelzw** [comment]
CI is green now
@isuruf this is ready for review

**Comment by xhochy** [comment]
@isuruf ping here. 

I am also happy to add myself as a maintainer if you want to avoid dealing with this package. Looks like this hasn't been touched for 6 years.

**Comment by isuruf** [comment]
This package is deprecated now. Use `winpthreads-devel` if you want to use a pthreads library.

**Comment by pavelzw** [comment]
> This package is deprecated now

this package is still used by mkl. we should update the corresponding packages

```
❯ pixi tree -i pthreads-win32 --platform win-64

pthreads-win32 2.9.1
└── libhwloc 2.11.1
    └── tbb 2021.13.0
        └── mkl 2024.1.0
            └── libblas 3.9.0
                ├── liblapack 3.9.0
                │   └── numpy 2.1.2
                ├── numpy 2.1.2 (*)
                └── libcblas 3.9.0
                    └── numpy 2.1.2 (*)
```

**Pull request merged**